tsl/test/expected/correlated_constraints.out

```diff
@@ -0,0 +1,285 @@
+SELECT * FROM _timescaledb_catalog.dimension_slice WHERE dimension_id = :dimension_id;
+ id | dimension_id |     range_start      |      range_end      
+----+--------------+----------------------+---------------------
+  3 |            2 | -9223372036854775808 | 9223372036854775807
+  4 |            2 |                    1 |                   9
+(2 rows)
+
+SELECT * FROM _timescaledb_catalog.chunk_constraint WHERE constraint_name LIKE '_$CC_con%' ORDER BY chunk_id;
+ chunk_id | dimension_slice_id |  constraint_name  | hypertable_constraint_name 
+----------+--------------------+-------------------+----------------------------
+        1 |                  4 | _$CC_constraint_4 | 
+        2 |                  3 | _$CC_constraint_3 | 
+(2 rows)
+
+-- A query using a WHERE clause on "sensor_id" column will scan the proper chunk
+-- due to chunk exclusion using correlated constraints ranges calculated above
+:PREFIX SELECT * FROM sample_table WHERE sensor_id > 9;
+          QUERY PLAN          
+------------------------------
+ Seq Scan on _hyper_1_2_chunk
+   Filter: (sensor_id > 9)
+(2 rows)
+
+-- Newly added chunks should also point to this MIN/MAX entry
+\set start_date '2024-01-28 01:09:51.583252+05:30'
+INSERT INTO sample_table VALUES (:'start_date'::timestamptz, 1, 9.6054, 78.999, 'new row4');
+SELECT * FROM _timescaledb_catalog.chunk_constraint WHERE dimension_slice_id = :slice_id ORDER BY chunk_id;
+ chunk_id | dimension_slice_id |  constraint_name  | hypertable_constraint_name 
+----------+--------------------+-------------------+----------------------------
+        2 |                  3 | _$CC_constraint_3 | 
+        5 |                  3 | _$CC_constraint_3 | 
+(2 rows)
+
+-- use the remove_dimension API to remove the correlated dimension entries
+SELECT * from remove_dimension('sample_table', 'sensor_id');
+ dimension_id | dropped 
+--------------+---------
+            2 | t
+(1 row)
+
+SELECT * FROM _timescaledb_catalog.dimension WHERE type = 'C';
+ id | hypertable_id | column_name | column_type | aligned | num_slices | partitioning_func_schema | partitioning_func | interval_length | compress_interval_length | integer_now_func_schema | integer_now_func | type 
+----+---------------+-------------+-------------+---------+------------+--------------------------+-------------------+-----------------+--------------------------+-------------------------+------------------+------
+(0 rows)
+
+SELECT * from remove_dimension('sample_table', 'sensor_id', true);
+NOTICE:  no correlated dimension on column "sensor_id", skipping
+ dimension_id | dropped 
+--------------+---------
+            0 | f
+(1 row)
+
+\set ON_ERROR_STOP 0
+SELECT * from remove_dimension('sample_table', 'sensor_id');
+ERROR:  no correlated dimension on column "sensor_id"
+-- should only work on correlated dimensions
+SELECT * from remove_dimension('sample_table', 'time');
+ERROR:  no correlated dimension on column "time"
```

Comment:
I think it would be more informational if the error for other type of dimensions was something like:

```
range dimension "time" cannot be removed
hint: Only correlated dimensions can be removed.
```

The current message is slightly confusing because the error assumes that the intention of the user was to remove a correlated dimension.

But just by looking at the API and the function name, it is implied that you can remove any dimensions (irrespective of type). So, if a user tries to intentionally remove the "time" dimension  (knowing it is a time dimension), the error talks about not finding a correlated dimension. Instead we should let the user know that removing "time"/range dimensions is not supported. 

As an aside, I think it would be easy to support removing dimensions if there are no chunks in the hypertable, which might be useful when you are creating the hypertable "schema" and erroneously adds a dimension. We don't need to implement that as part of this PR. Just saying it is easy to do.

Reply:
@erimatnor the errhint does elaborate that 

"remove_dimension can only be used to remove correlated dimension constraints."

Since it can be called on any column, we will have to seek metadata to further identify the dimension type of the column and then say that it cannot be removed..

We will have to ensure that we add documentation for `remove_dimension` which mentions the restriction on this currently working on correlated constraints only.